### ISSUE TITLE
NotSerializableException when using Gradle Reckon plugin

### ISSUE THREAD

**Issue by Riggs333** [asciidoctor/asciidoctor-gradle-plugin#293]
Using the Asciidoctor Gradle plugin in combination with the [Reckon Plugin](https://github.com/ajoberstar/reckon)
```groovy
plugins {
    id 'org.ajoberstar.reckon' version '0.9.0'
    id 'org.asciidoctor.jvm.convert' version '2.0.0-rc.1'
}
```

fails with 

```java
* What went wrong:
Execution failed for task ':asciidoctor'.
> java.io.NotSerializableException: org.ajoberstar.reckon.gradle.ReckonPlugin$DelayedVersion
```

Not sure if this is an issue or can be solved in the Reckon Plugin.

See here for a minimal sample project:
https://github.com/Riggs333/gradle-asciidoctor-with-reckon

### Stacktrace
```java
...
Caused by: java.io.NotSerializableException: org.ajoberstar.reckon.gradle.ReckonPlugin$DelayedVersion
        at org.asciidoctor.gradle.internal.ExecutorConfigurationContainer$_toFile_closure1$_closure2.doCall(ExecutorConfigurationContainer.groovy:46)
        at org.asciidoctor.gradle.internal.ExecutorConfigurationContainer$_toFile_closure1$_closure2.call(ExecutorConfigurationContainer.groovy)
        at org.asciidoctor.gradle.internal.ExecutorConfigurationContainer$_toFile_closure1.doCall(ExecutorConfigurationContainer.groovy:44)
        at org.asciidoctor.gradle.internal.ExecutorConfigurationContainer$_toFile_closure1.call(ExecutorConfigurationContainer.groovy)
        at org.asciidoctor.gradle.internal.ExecutorConfigurationContainer.toFile(ExecutorConfigurationContainer.groovy:43)
        at org.asciidoctor.gradle.internal.JavaExecUtils.writeExecConfigurationData(JavaExecUtils.groovy:74)
        at org.asciidoctor.gradle.jvm.AbstractAsciidoctorTask.runWithJavaExec(AbstractAsciidoctorTask.groovy:898)
        at org.asciidoctor.gradle.jvm.AbstractAsciidoctorTask.processAsciidocSources(AbstractAsciidoctorTask.groovy:565)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:103)
        ... 75 more
```

**Comment by ysb33r** [comment]
Thanks. Will have a look.

**Comment by jagedn** [comment]
The issue come because the `ajoberstar` delay the resolution of the version into a non serializable object

As the `asciidoctor` plugin try to use the `version` project attribute in the attribute map for `project-version` and `revnumber`  it fails.

I solved this issue in this way:

asciidoctor{
    sources{
        include 'index.adoc'
    }
    String v = "$project.version"
    attributes "project-version" : v, "revnumber":v
}

**Comment by ysb33r** [comment]
This is actually an interesting use-case. Most people, including myself, have always just assumed `project.version` to be a string. I am going to add something in the code to deal with this.

**Comment by ysb33r** [comment]
I am closing this as there will be fix in the final 2.0 release schedules (hopefully) for the week of 25 March.
